packages/mui-utils/src/getReactNodeRef/getReactNodeRef.ts

```diff
@@ -8,13 +8,9 @@ import * as React from 'react';
  * @returns React.Ref<any> | null
  */
 export default function getReactNodeRef(element: React.ReactNode): React.Ref<any> | null {
-  if (!element || !React.isValidElement(element)) {
-    return null;
-  }
-
```

Comment:
@aarongarciah rebased PR and removed these checks as it was decided here https://github.com/mui/material-ui/pull/43022#discussion_r1687208774

Reply:
The comment on [line 15](https://github.com/mui/material-ui/pull/43022/files#diff-2aabece439eee5bb9309cc46b1b3c3b690ee8a7231cbac1ba495cf832e506152R15) needs to be updated since we're not using `React.isValidElement` anymore.

> // We cannot check for it, but isValidElement is true at this point

Reply:
Also the JSDoc comment probably needs to be updated:

> It will return null if the node is not a valid React element.

This is not true since the function will error if "non-object" types (which are part of `React.ReactNode`) are passed to it (null, undefined, string, boolean, etc.)